File: pegasusX/apps/warehouse-app-android/app/src/main/java/com/pegasusx/warehouse/ui/screens/operations/OperationsPricingPreview.kt
package com.pegasusx.warehouse.ui.screens.operations

import androidx.compose.ui.res.stringResource

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import com.pegasusx.warehouse.data.model.RetailerOverridePreview
import com.pegasusx.warehouse.ui.theme.PegasusSpacing
import com.pegasusx.warehouse.ui.components.WarehouseSectionTitle

@Composable
fun OperationsPricingPreview(
    productId: String,
    onProductIdChange: (String) -> Unit,
    retailerId: String,
    onRetailerIdChange: (String) -> Unit,
    proposedPrice: String,
    onProposedPriceChange: (String) -> Unit,
    previewLoading: Boolean,
    preview: RetailerOverridePreview?
) {
    Column {
        WarehouseSectionTitle("Pricing impact preview (read-only)")
        Text(
            "Preview how a proposed retailer price would compare to catalog list price. Does not create overrides.",
            style = MaterialTheme.typography.bodySmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
        )
        OutlinedTextField(
            value = productId,
            onValueChange = onProductIdChange,
            label = { Text("Product / SKU ID") },
            modifier = Modifier.fillMaxWidth(),
            singleLine = true,
        )
        OutlinedTextField(
            value = retailerId,
            onValueChange = onRetailerIdChange,
            label = { Text("Retailer ID (optional)") },
            modifier = Modifier.fillMaxWidth(),
            singleLine = true,
        )
        OutlinedTextField(
            value = proposedPrice,
            onValueChange = onProposedPriceChange,
            label = { Text("Proposed price (minor units)") },
            modifier = Modifier.fillMaxWidth(),
            singleLine = true,
            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
        )
        if (previewLoading) {
            Text("Loading preview…", style = MaterialTheme.typography.bodySmall)
        }
        preview?.let { p ->
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(vertical = PegasusSpacing.sm),
                verticalArrangement = Arrangement.spacedBy(4.dp),
            ) {
                Text(stringResource(R.string.mobile_warehouse_ui_retailers_on_sku_retailersonskucount, p.retailersOnSkuCount))
                Text(stringResource(R.string.mobile_warehouse_ui_active_overrides_activeoverridecount, p.activeOverrideCount))
                Text(stringResource(R.string.mobile_warehouse_ui_catalog_list_price_cataloglistprice, p.catalogListPrice))
                Text(stringResource(R.string.mobile_warehouse_ui_margin_delta_unit_margindeltaperunit, p.marginDeltaPerUnit))
                Text(
                    p.marginEstimateLabel,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
                if (p.readOnly == true) {
                    Text(
                        "Read-only — contact supplier to apply overrides.",
                        style = MaterialTheme.typography.bodyMedium,
                    )
                }
            }
        }
    }
}

File: pegasusX/apps/warehouse-app-ios/WarehouseApp/Views/Operations/OperationsPricingPreview.swift
import SwiftUI

struct OperationsPricingPreview: View {
    @Binding var productId: String
    @Binding var retailerId: String
    @Binding var proposedPrice: String
    
    let previewLoading: Bool
    let preview: RetailerOverridePreview?
    
    let onSchedulePreview: () -> Void
    
    var body: some View {
        Group {
            Section {
                WarehouseSectionHeader(
                    title: "Pricing impact preview (read-only)",
                    subtitle: "Preview proposed retailer price vs catalog list price. Does not create overrides."
                )
            }
            Section {
                TextField("warehouse_portal.residual.text.product_sku_id", text: $productId)
                    .textInputAutocapitalization(.never)
                    .autocorrectionDisabled()
                    .onChange(of: productId) { _, _ in onSchedulePreview() }
                TextField("warehouse_portal.residual.text.retailer_id_optional", text: $retailerId)
                    .textInputAutocapitalization(.never)
                    .autocorrectionDisabled()
                    .onChange(of: retailerId) { _, _ in onSchedulePreview() }
                TextField("warehouse_portal.residual.text.proposed_price_minor_units", text: $proposedPrice)
                    .keyboardType(.numberPad)
                    .onChange(of: proposedPrice) { _, _ in onSchedulePreview() }

                if previewLoading {
                    Text("warehouse_portal.operations.operations_pricing_preview.text.loading_preview")
                        .font(.footnote)
                        .foregroundStyle(.secondary)
                }
                if let preview {
                    LabeledContent("Retailers on SKU", value: "\(preview.retailersOnSkuCount)")
                    LabeledContent("Active overrides", value: "\(preview.activeOverrideCount)")
                    LabeledContent("Catalog list price", value: "\(preview.catalogListPrice)")
                    LabeledContent("Margin delta / unit", value: "\(preview.marginDeltaPerUnit)")
                    Text(preview.marginEstimateLabel)
                        .font(.footnote)
                        .foregroundStyle(.secondary)
                    if preview.readOnly == true {
                        Text("warehouse_portal.operations.operations_pricing_preview.text.read_only_contact_supplier_to_apply_overrides")
                            .font(.footnote.weight(.medium))
                    }
                }
            }
        }
    }
}
